app/src/main/java/com/acsbendi/requestinspectorwebview/RequestInspectorJavaScriptInterface.kt

```diff
@@ -17,10 +18,10 @@ internal class RequestInspectorJavaScriptInterface(webView: WebView) {
 
     private val recordedRequests = ArrayList<RecordedRequest>()
 
-    fun findRecordedRequestForUrl(url: String): RecordedRequest? {
+    fun findRecordedRequestForUrl(url: String, method: String): RecordedRequest? {
         return synchronized(recordedRequests) {
             recordedRequests.find { recordedRequest ->
-                url.contains(recordedRequest.url)
+                url == recordedRequest.url && method == recordedRequest.method
             }
```

Comment:
The reason why `contains` was used instead of `==` when checking the URL match is that sometimes it fails to record the whole URL, and we want to be able to find it based on a partial match. So I suggest we keep the `contains` for the URL.